packages/debugger/lib/data/reducers.js

```diff
@@ -49,9 +54,12 @@ function scopes(state = DEFAULT_SCOPES, action) {
 
             variables: [
               ...variables,
-
               { name: action.node.name, id: action.node.id }
-            ]
+            ],
+
+            visibles: visible
+              ? [...visibles, { name: action.node.name, id: action.node.id }]
+              : visibles
```

Comment:
How do you feel about modifying `variables` to include `visibility` (maybe even the full-form, not just`true`/`false`!), and then just filtering in the selector?

Reply:
Hm.  Now that you bring it up, I'm realizing this can be done even more simply, without keeping track of any extra information at all.  Because we can filter in the selector just by looking up the visibility in the variable definition, because we have the variable ID and all the definitions!  I think I'm going to go back and change it to do it that way, reverting the reducer to how it was and having all the work occur in the selector.